File: sample-code/storage-app/azure/presignedURLs.py
# need to pip install azure-storage-blob

from datetime import datetime, timedelta
from azure.storage.blob import BlobServiceClient, generate_account_sas, ResourceTypes, AccountSasPermissions

storage_account_name = ""
account_access_key = ""
storage_container_name=""
blob_path=""

sas_token = generate_account_sas(
    account_name=storage_account_name,
    account_key=account_access_key,
    resource_types=ResourceTypes(object=True),
    permission=AccountSasPermissions(read=True),
    expiry=datetime.utcnow() + timedelta(hours=1)
)

the_sas_url = f"https://{storage_account_name}.blob.core.windows.net/{storage_container_name}/{blob_path}?{sas_token}"
print(the_sas_url)

File: sample-code/storage-app/azure/presignedURLs.js
//Need to npm install @azure/storage-blob
var storage = require("@azure/storage-blob")
const accountname ="";
const key = "";
const cerds = new storage.StorageSharedKeyCredential(accountname,key);
const blobServiceClient = new storage.BlobServiceClient(`https://${accountname}.blob.core.windows.net`,cerds);
const containerName="test";
const client =blobServiceClient.getContainerClient(containerName)
const blobName="help.txt";
const blobClient = client.getBlobClient(blobName);

const blobSAS = storage.generateBlobSASQueryParameters({
    containerName,
    blobName,
    permissions: storage.BlobSASPermissions.parse("racwd"),
    startsOn: new Date(),
    expiresOn: new Date(new Date().valueOf() + 86400)
  },
  cerds
).toString();

const sasUrl= blobClient.url+"?"+blobSAS;
console.log(sasUrl);
